packages/project/src/android/project.ts

```diff
@@ -49,11 +48,60 @@ export class AndroidProject {
     return null;
   }
 
+  /**
+   * Update the Android package name. This renames the package in `AndroidManifest.xml`,
+   * the `applicationId` in `app/build.gradle`, and renames the java
+   * package for the `MainActivity.java` file.
+   * 
+   * This action will mutate the project on disk!
+   */
   async setPackageName(packageName: string) {
+    const oldPackageName = await this.manifest.getDocumentElement()?.getAttribute('package');
     this.manifest.getDocumentElement()?.setAttribute('package', packageName);
     await this.appBuildGradle?.setApplicationId(packageName);
+
+    if (!this.getAppRoot()) {
+      return;
+    }
+
+    const oldPackageParts = oldPackageName?.split('.') ?? [];
+    const newPackageParts = packageName.split('.');
+
+    const sourceDir = join(this.getAppRoot()!, 'src', 'main', 'java');
+    const destDir = join(sourceDir, ...newPackageParts);
+
+    let activityFile = join(sourceDir, ...oldPackageParts, 'MainActivity.java');
+
+    // Make the new directory tree and any missing parents
+    await mkdirp(destDir);
+    // Move the old activity file over
+    await move(activityFile, join(destDir, 'MainActivity.java'));
```

Comment:
Maybe we should handle a case where the previous and current packageId is the same (right now it throws an error). I think it would be useful when a user changes something in the configuration but the packageId stays unchanged.